packages/media-utils/src/components/media-upload/index.js

```diff
@@ -86,6 +165,43 @@ const getGalleryDetailsMediaFrame = () => {
 	} );
 };
 
+/**
+ * Prepares the edit state and sets the listener for "Edit Image" button click.
+ *
+ * @param {Object} types passes the allowedTypes object to the media library.
+ * @return {wp.media.view.MediaFrame.Select} The default media workflow.
+ */
+const addEditState = ( types ) => {
```

Comment:
Curious if we could move this into core directly, rather than extending the Select frame only for Gutenberg to include the edit state?

Reply:
Thanks for looking at this! I explored doing it globally in the select frame but found it difficult to set the context of `allowedTypes` and selections from this component. The way that we are using frames in Gutenberg is a bit different than we do with core because we base everything off of one default frame. If we set it globally in the select frame, we would have conflict with the Video block because we use the default frame with mime types passed from the props. In this function, we need to pass the allowed mimetypes to the `createStates` function as `types` and as such, each frame (featured, gallery, default) has a specific state to get the selection from. (library, gallery, featured-image) I think where I found the hangup is passing `allowedTypes` from Gutenberg to the Core code. I could be missing something so please let me know if you have any ideas how :) I also want to be cautious about changing Core code and breaking the Classic Editor plugin's flow

Reply:
Have it working in the Core `Select` ! I need to explore one last bug where the select button is blank when clicking "back" from the edit screen. This addEditState function can now be removed from this PR. The featured image and gallery states will still need this PR to fix it in Gutenberg. Thanks for calling this out @joemcgill !

Reply:
> Have it working in the Core `Select` !

This is great news! Did you manage to resolve it in Core fully?

Reply:
@gziolo Yeah! I think we are good now. I've updated this PR to remove the default edit state handling. Now the edit state is added to select using the latest patch in https://core.trac.wordpress.org/ticket/48028

With the core patch above **and** this PR applied to a local Gutenberg, the issue is fixed in gallery, featured image, image block, and inline image block. (may npm have mercy on your computer while testing)

I also removed a declaration in the `frameConfig` to not define a button text for the default state. It is redundant with the core default select button text. Also it caused conflict when hitting the back button once in the `Edit Image` frame.

Reply:
and to be clear. The Core side is resolved fully once the patch is committed, however, this PR is needed in order for folks using the new editor to reap the benefits. :) 

New commit incoming to resolve Travis errors.